sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -394,13 +395,14 @@ case class DropTable(
     ifExists: Boolean) extends Command
 
 /**
- * The logical plan of the ALTER TABLE command that works for v2 tables.
+ * The base class for ALTER TABLE commands that work for v2 tables.
  */
-case class AlterTable(
-    catalog: TableCatalog,
-    ident: Identifier,
-    table: NamedRelation,
-    changes: Seq[TableChange]) extends Command {
+abstract class AlterTable extends Command {
+  def table: LogicalPlan
+
+  def changes: Seq[TableChange]
+
+  override def children: Seq[LogicalPlan] = Seq(table)
 
   override lazy val resolved: Boolean = table.resolved && {
     changes.forall {
```

Comment:
Currently, `changes` will be triggered even though the resolved table is `V1Table` because `ResolveSessionCatalog` runs at the end (`resolved` is called before `V1Table` is handled). Should we check if table is not `V1Table` here or extra computation is fine for simplicity?

Reply:
I think it's fine to remove it completely. This is duplicated with the logic in `CheckAnalysis`.

Reply:
Great, thanks! (I added a missing case for `UpdateColumnPosition`)